addons/l10n_it_edi/models/ir_actions_report.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+
+import io
+
+from odoo import models
+from odoo.tools.pdf import OdooPdfFileReader, OdooPdfFileWriter
+
+
+class IrActionsReport(models.Model):
+    _inherit = 'ir.actions.report'
+
+    def _render_qweb_pdf_prepare_streams(self, report_ref, data, res_ids=None):
+        # EXTENDS account
+        if (collected_streams := super()._render_qweb_pdf_prepare_streams(report_ref, data, res_ids=res_ids)
+            and res_ids
+            and len(res_ids) == 1
+            and self._get_report(report_ref).report_name in ('account.report_invoice_with_payments', 'account.report_invoice')
+            and (move := self.env['account.move'].browse(res_ids[0]))):
+
+            if move.is_sale_document() and move.state != 'draft':
+                pdf_stream = collected_streams[move.id]['stream']
+
+                # Read pdf content.
+                pdf_content = pdf_stream.getvalue()
+                reader_buffer = io.BytesIO(pdf_content)
+                reader = OdooPdfFileReader(reader_buffer, strict=False)
+
+                # Post-process and embed the additional files.
+                writer = OdooPdfFileWriter()
+                writer.cloneReaderDocumentRoot(reader)
+
+                # Add the attachments to the pdf file
+                for attachment in move.sudo().attachment_ids.filtered(lambda x: x._l10n_it_edi_check_processable()):
+                    # The attachements on the edi documents are only system readable
+                    # because they don't have res_id and res_model, here we are sure that
+                    # the user has access to the move and edi document
+                    writer.embed_odoo_attachment(attachment)
```

Comment:
Since when do we add the XML in the PDF for IT?

Reply:
I don't know... two years ? 
https://github.com/odoo/odoo/blame/aa075760986644c0fb0f89de1d7955dff286a442/addons/l10n_it_edi/models/account_edi_format.py

![image](https://github.com/odoo/odoo/assets/1665365/4749945d-9523-4a10-aa2f-6939fc6ba721)

More? 
![image](https://github.com/odoo/odoo/assets/1665365/78c01bb4-081b-409c-8066-83767986a735)

Reply:
I know it sounds stupid. It's also stupid to me. 
I don't even know if it works. Want me to remove that?

Reply:
Done.